src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -774,7 +774,11 @@ private PackageSpec GetPackageSpec(IMSBuildProject project, IReadOnlyDictionary<
             restoreMetadata.CacheFilePath = NoOpRestoreUtilities.GetProjectCacheFilePath(outputPath, project.FullPath);
             restoreMetadata.ConfigFilePaths = settings.GetConfigFilePaths();
             restoreMetadata.OutputPath = outputPath;
-            restoreMetadata.OriginalTargetFrameworks.AddRange(targetFrameworkInfos.Select(e => e.TargetAlias));
```

Comment:
Personally I don't like `.ForEach`. If remember correctly this one let targetFrameworkInfos array mutate from inside .ForFach loop. Not matter much for this PR but in general better to use normal `foreach`.

Reply:
I don't think there's be a function different than a `foreach` loop. 
I am populating a property of the type. 
The `ForEach` here allows for better readability because of the rest of the code in this method is written. 

I can reconsider if people care.

Reply:
I don't have a problem with `.ForEach`, but given the old code was already using `restoreMetadata.OriginalTargetFrameworks.AddRange`, I think that's a nicer option, Something like:

```cs
restoreMetadata.OriginalTargetFrameworks.AddRange(targetFrameworkInfos.Select(tfi => !string.IsNullOrEmpty(tfi.TargetAlias ? tfi.TargetAlias : tfi.FrameworkName.GetShortFolderName()));
```